pkg/resource/generator.go

```diff
@@ -19,6 +19,11 @@ import (
 	"github.com/openshift/cluster-image-registry-operator/pkg/parameters"
 )
 
+type ChecksumInput struct {
+	Spec   regopapi.ImageRegistrySpec
```

Comment:
If this is a generic `struct` then the field type should be a `interface{}`. If this `struct` is only for `ImageRegistry` then please rename it to `ImageRegistryChecksumInput` or something.

Reply:
renamed.